cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/ScheduleSpecificationCodec.java

```diff
@@ -32,11 +34,48 @@
  *
  */
 public class ScheduleSpecificationCodec extends AbstractSpecificationCodec<ScheduleSpecification> {
+
+  /**
+   * Schedule Type.
+   */
+  private enum ScheduleType {
+    ORIGINAL_TIME,
```

Comment:
``TIME`` assumes that the object is a ``TimeSchedule``, whereas ``ORIGINAL_TIME`` assumes that we're dealing with a ``Schedule`` object in which ``cronEntry`` is not null.